plugins/src/hashlookup/select.go

```diff
@@ -0,0 +1,241 @@
+	}
+
+	fields := append(leftStr, rightStr...)
+	if topLevel {
+		fields = append(fields, [2]string{"operator", "and"})
+	}
+
+	return fields, nil
+}
+
+func handleSelectWhereOrExpr(expr *sqlparser.Expr, topLevel bool, parent *sqlparser.Expr) ([][2]string, error) {
+	orExpr := (*expr).(*sqlparser.OrExpr)
+	leftExpr := orExpr.Left
+	rightExpr := orExpr.Right
+
+	leftStr, err := handleSelectWhere(&leftExpr, false, expr)
+	if err != nil {
+		return nil, err
+	}
+
+	rightStr, err := handleSelectWhere(&rightExpr, false, expr)
+	if err != nil {
+		return nil, err
+	}
+
+	fields := append(leftStr, rightStr...)
+	if topLevel {
+		fields = append(fields, [2]string{"operator", "or"})
+	}
+
+	return fields, nil
+}
+
+// Between a and b.
+// The meaning is equal to range query
+func handleSelectWhereBetweenExpr(expr *sqlparser.Expr, topLevel bool) ([][2]string, error) {
+	rangeCond := (*expr).(*sqlparser.RangeCond)
+	colName, ok := rangeCond.Left.(*sqlparser.ColName)
+
+	if !ok {
+		return nil, errors.New("Range column name missing")
+	}
+
+	//colNameStr := sqlparser.String(colName)
+	colNameStr := strings.Trim(sqlparser.String(colName), "`")
+	//fromStr := strings.Trim(sqlparser.String(rangeCond.From), "'")
+	//toStr := strings.Trim(sqlparser.String(rangeCond.To), "'")
+
+	var from string
+	var to string
+
+	// Prepare a 'From' value
+	switch expr := rangeCond.From.(type) {
+	case *sqlparser.SQLVal:
+		switch expr.Type {
+		case sqlparser.IntVal, sqlparser.FloatVal, sqlparser.StrVal:
+			from = string(expr.Val)
+		default:
+			return nil, fmt.Errorf("Invalid BETWEEN 'from' value: %v (type %v)", string(expr.Val), expr.Type)
+		}
+	default:
+		return nil, fmt.Errorf("Invalid BETWEEN 'from' value: %v", strings.Trim(sqlparser.String(rangeCond.From), "'"))
+	}
+
+	// Prepare a 'To' value
+	switch expr := rangeCond.To.(type) {
+	case *sqlparser.SQLVal:
+		switch expr.Type {
+		case sqlparser.IntVal, sqlparser.FloatVal, sqlparser.StrVal:
+			to = string(expr.Val)
+		default:
+			return nil, fmt.Errorf("Invalid BETWEEN 'to' value: %v (type %v)", string(expr.Val), expr.Type)
+		}
+	default:
+		return nil, fmt.Errorf("Invalid BETWEEN 'to' value: %v", strings.Trim(sqlparser.String(rangeCond.To), "'"))
+	}
+
+	// Build resulting fields list
+	var fields [][2]string
+	fields = append(fields, [2]string{colNameStr + "_from", from})
+	fields = append(fields, [2]string{colNameStr + "_to", to})
+
+	return fields, nil
+}
+
+func handleSelectWhereParenExpr(expr *sqlparser.Expr, topLevel bool, parent *sqlparser.Expr) ([][2]string, error) {
+	parentBoolExpr := (*expr).(*sqlparser.ParenExpr)
+	boolExpr := parentBoolExpr.Expr
+
+	// If parent is the top level, bool must is needed
+	var isThisTopLevel = false
+	if topLevel {
+		isThisTopLevel = true
+	}
+
+	return handleSelectWhere(&boolExpr, isThisTopLevel, parent)
+}
+
+func buildNestedFuncStrValue(nestedFunc *sqlparser.FuncExpr) (string, error) {
+	return "", errors.New("Unsupported function: " + nestedFunc.Name.String())
+}
+
```

Comment:
/*
 * Check the right part of the expression
 * and return its value of specific type.
 *
 * Receives SQL expression to process
 */